src/pages/MainPage.jsx

```diff
@@ -0,0 +1,31 @@
+// src/pages/MainPage.jsx
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+function MainPage() {
+  const [concerts, setConcerts] = useState([]);
+
+  useEffect(() => {
+    fetch('https://book-a-concert-api.onrender.com/concerts')
+      .then(response => response.json())
+      .then(data => setConcerts(data));
+  }, []);
+
+  return (
+    <div className="main-page">
+      <h1>Available Concerts</h1>
+      <div className="concert-list">
+        {concerts.map(concert => (
+          <Link key={concert.id} to={`/concerts/${concert.id}`}>
```

Comment:
You might consider using `<ul><li></li></ul>` element here instead of div, that would improve semantics